### DISCUSSION TITLE
Require some help migrating to Nginx please

### DISCUSSION THREAD

**Discussion by bobokun** [TandoorRecipes/recipes#341] [Q&A] [answered]
Hi, I'm hoping to get some help with migrating media serving from gunicorn to nginx using Unraid. I've been trying to follow the instructions found [here](https://vabene1111.github.io/recipes/install/docker/) however I'm not sure what is wrong with my config and also I'm not sure where to add the these environment variables to or what needs to be entered there. Is is like this?

VIRTUAL_HOST= recipes.mydns.com
LETSENCRYPT_HOST= recipes.mydns.com
LETSENCRYPT_EMAIL= myemail@mydns.com

Here is my configuration for my containers

**Container name: recipes
Repository: vabene1111/recipes**
![image](https://user-images.githubusercontent.com/12660469/104961099-a81dd200-59a3-11eb-84bb-c0ec0c7cc6ef.png)

Container name: nginx_recipes
Repository: linuxserver/nginx
![image](https://user-images.githubusercontent.com/12660469/104961196-d4d1e980-59a3-11eb-9930-cd423071b4ea.png)

Recipes.conf
```
server {
  listen 80;
  server_name localhost;

  client_max_body_size 16M;

  # serve media files
  location /media/ {
    alias /media/;
  }
  # pass requests for dynamic content to gunicorn
  location / {
	
    proxy_set_header Host $host;
    proxy_pass http://recipes:8080;
  }
}

```
Going to system version everything seems to be okay but I don't see any of my images.
![image](https://user-images.githubusercontent.com/12660469/104961334-1b274880-59a4-11eb-8db7-8095ecacdf40.png)

![image](https://user-images.githubusercontent.com/12660469/104961597-a56fac80-59a4-11eb-86cb-7d60322e0943.png)

**Comment by vabene1111** [comment]
Hmm this looks pretty correct to me 🤔 
I am not very familiar with the unraid setup tough .. maybe @CorneliousJD can help, i know he answers some questions over on the unraid forum ...

**Reply by CorneliousJD** [reply]
> Hmm this looks pretty correct to me 🤔
> I am not very familiar with the unraid setup tough .. maybe @CorneliousJD can help, i know he answers some questions over on the unraid forum ...

I actually directed them here because I wasn't able to provide the help required to get this working, and would personally just leave it on Guincorn serving the media to keep it simple for the unraid container. 

Sorry!

**Reply by vabene1111** [reply]
no worries, you are not obligated to provide support for anything, just tought you might already know the solution :) thanks btw for helping so many people

**Comment by bobokun** [comment]
I'm hoping @stormsz can help chime in too since he created the initial commit for the recipes reverse proxy on the swag container. I assume you got it working?

**Reply by Foxhound4444** [reply]
I'm having this same issue. I've got everything working on Swag and using postgresql but the images aren't loading. I'm using the same config as @bobokun . 

When I set gunicorn to 1, the images load when i load the webui via my local IP address. But when i use my reverse proxy, it doesn't seem to work.

**Comment by vabene1111** [comment]
using gunciorn as the media server should work no matter if there are any reverse proxies in front of it or not. 

As i do now have an unraid setup and @CorneliousJD who is very skilled with it and also know this application quite well cannot help you i fear that i do not know how to provide further support. 

I can just generally say that media files are stored inside a mediafiles directory (which looks like you mounted correctly) and that those files need to be served at http://recipedomain.tld/media/recipes/filename.png (i just wrote that here without looking it up, take a look at the broken image for the exact path)

**Reply by Foxhound4444** [reply]
Thanks for the reply! I checked the path of the images and they appear to be correct and what you described. Hope to help fix this any way i can. Will keep playing around with this for now

**Reply by bobokun** [reply]
How did you get your reverse proxy to work using swag? I haven't been able to successfully get my reverse proxy working. It just returns me Bad Request (400). Can you share your recipes.subdomain.conf please?

**Reply by Foxhound4444** [reply]
@bobokun sure, i edited it a little to the following (please excuse the copy and paste. I have no idea what im doing to be honest):

> server {
>   listen 443;
>   listen [::]:443 ssl;
>   server_name recipes.*;
> 
>   client_max_body_size 16M;
> 
>   # serve media files
>   location /media/ {
>     alias /media/;
>   }
>   # pass requests for dynamic content to gunicorn
>   location / {
>     proxy_set_header Host $host;
>     proxy_pass https://recipes:8080;
>   }
> }

**Reply by bobokun** [reply]
When pasting the code you can highlight the entire code and press the insert code button to format it properly.
![image](https://user-images.githubusercontent.com/12660469/105098956-d9b49d00-5a78-11eb-8b47-f9ccd1ee53e1.png)

**Reply by Foxhound4444** [reply]
I put it in a text file and attached it to here for an easier approach
[recipes.subdomain.txt](https://github.com/vabene1111/recipes/files/5838862/recipes.subdomain.txt)

**Reply by bobokun** [reply]
hmm weird, even with your config it doesn't seem to be working for me. You placed it in your appdata\swag\nginx\proxy-confs right?

**Reply by Foxhound4444** [reply]
Yup. Is your error the 500 error? Check your log of your recipes docker and see if there's an error when trying to connect to the postgres database. I had to manually go in and give the user i created for the recipe database, superuser access. My log showed for a while that it couldn't connect to the postgresql db. After fixing that, it was able to connect but didn't have rights it needed. So i made the user a superuser as the .log was telling me to do. And boom, that fixed it and no more 500 error. LMK what your log tells you regarding connecting to your postgres - it may be due to that.

**Reply by bobokun** [reply]
I'm just seeing constant timeout .

```
[2021-01-19 22:24:18 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:12)
[2021-01-19 22:24:18 +0000] [12] [INFO] Worker exiting (pid: 12)
[2021-01-19 22:24:18 +0000] [13] [INFO] Booting worker with pid: 13
[2021-01-19 22:24:50 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:13)
[2021-01-19 22:24:50 +0000] [13] [INFO] Worker exiting (pid: 13)
[2021-01-19 22:24:50 +0000] [14] [INFO] Booting worker with pid: 14
[2021-01-19 22:24:50 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:13)
[2021-01-19 22:24:50 +0000] [13] [INFO] Worker exiting (pid: 13)
[2021-01-19 22:24:50 +0000] [14] [INFO] Booting worker with pid: 14
```

**Reply by Foxhound4444** [reply]
That is odd. That's what mine looks like without the worker timeout. Is there anything above this that shows the docker connecting to postgres?

**Reply by Foxhound4444** [reply]
@bobokun also i had to change my postgresql docker set to "host" instead of "bridge" so that the database can be reached locally by the recipe container. Otherwise, the postgresql docker didn't have any IP address. Then in your Recipe docker, set your "PostgreSQL Host" to your local ip address for your server (192.168.1.100) and leave the PostgreSQL port the same as long as you haven't changed that.

**Reply by bobokun** [reply]
yeah it can connect to PostgreSQL just fine, everything works perfectly locally but times out when trying to reach it through my subdomain reverse proxy

**Reply by Foxhound4444** [reply]
Is Swag loading the .config file in the log? Mine wasn't in the beginning for some reason. Had to change from using swag's config for recipes to the one listed here in the github (i added the 443 port at the top as you can see in the file i attached earlier)

**Comment by bobokun** [comment, marked as answer]
Okay I got it working!! I've attached my working config for swag.
[recipes.subdomain.conf.txt](https://github.com/vabene1111/recipes/files/5839460/recipes.subdomain.conf.txt)

One thing you need to add in your swag container is a mapping to your media/recipes folder
![image](https://user-images.githubusercontent.com/12660469/105111951-045e2000-5a90-11eb-8b90-4d4bc1f0cac8.png)

Also one thing to note is I couldn't get it to work until I changed my Allowed Hosts key to * 
![image](https://user-images.githubusercontent.com/12660469/105112294-b7c71480-5a90-11eb-8418-73413e3f5b8c.png)
When I added my subdomain and my IP address it didn't work. Only * allowed my reverse proxy to work.

I hope this helps! Thank you for everyone who contributed to helping!

**Reply by Foxhound4444** [reply]
Thank you for sharing the fix!! Got mine properly working as well :)

**Reply by vabene1111** [reply]
If you want feel free to open a PR to the docs to explain how to setup an additional reverse proxy for the unraid image :)

otherwise i will just link this conversation which is probably already quite good.
